pandas/tests/tslibs/test_conversion.py

```diff
@@ -80,6 +86,12 @@ def test_ensure_datetime64ns_bigendian():
     tm.assert_numpy_array_equal(result, expected)
 
 
+def test_ensure_timedelta64ns_overflows():
+    arr = np.arange(10).astype("m8[Y]") * 100
+    with pytest.raises(OutOfBoundsDatetime, match="Out of bounds"):
```

Comment:
this is the wrong type that is raises for this dtypes

Reply:
yah, im not too bothered by the exception type, but the exception message is going to have a datetime string in it that wont be helpful.  giving this some thought